File: MailMergeAndReporting/MailMergeAndReportingExamples.cs
using Helpers;
using NUnit.Framework;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Assert = NUnit.Framework.Assert;
using Description = Microsoft.VisualStudio.TestTools.UnitTesting.DescriptionAttribute;

namespace Examples.MailMergeAndReporting
{
    [TestClass, TestFixture]
    public class MailMergeAndReportingExamples
    {	
        [TestMethod, Test, Owner("Console")]
        public void ApplyCustomLogicToEmptyRegions()
        {
            TestHelper.SetDataDir("MailMergeAndReporting/ApplyCustomLogicToEmptyRegions");

            ApplyCustomLogicToEmptyRegionsExample.Program.Main();
        }

        [TestMethod, Test, Owner("Console")]
        public void XMLMailMerge()
        {
            TestHelper.SetDataDir("MailMergeAndReporting/XMLMailMerge");

            XMLMailMergeExample.Program.Main();
        }

        [TestMethod, Test, Owner("Console")]
        public void RemoveEmptyRegions()
        {
            TestHelper.SetDataDir("MailMergeAndReporting/RemoveEmptyRegions");

            RemoveEmptyRegionsExample.Program.Main();
        }

        [TestMethod, Test, Owner("Console")]
        public void NestedMailMerge()
        {
            TestHelper.SetDataDir("MailMergeAndReporting/NestedMailMerge");

            NestedMailMergeExample.Program.Main();
        }

        [TestMethod, Test, Owner("Console")]
        public void MultipleDocsInMailMerge()
        {
            TestHelper.SetDataDir("MailMergeAndReporting/MultipleDocsInMailMerge");

            MultipleDocsInMailMergeExample.Program.Main();
        }

        [TestMethod, Test, Owner("Console")]
        public void MailMergeFormFields()
        {
            TestHelper.SetDataDir("MailMergeAndReporting/MailMergeFormFields");

            MailMergeFormFieldsExample.Program.Main();
        }

        [TestMethod, Test, Owner("Console")]
        public void LINQtoXMLMailMerge()
        {
            TestHelper.SetDataDir("MailMergeAndReporting/LINQtoXMLMailMerge");

            LINQtoXMLMailMergeExample.Program.Main();
        }

    }
}

File: MailMergeAndReporting/MailMergeAndReportingExamples.vb
Imports Helpers
Imports NUnit.Framework
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports Assert = NUnit.Framework.Assert
Imports Description = Microsoft.VisualStudio.TestTools.UnitTesting.DescriptionAttribute

Namespace Examples.MailMergeAndReporting
	<TestClass, TestFixture> _
	Public Class MailMergeAndReportingExamples
        <TestMethod(), Test(), Owner("Console")> _
	        Public Sub ApplyCustomLogicToEmptyRegions()
	            TestHelper.SetDataDir("MailMergeAndReporting/ApplyCustomLogicToEmptyRegions")
	
	            ApplyCustomLogicToEmptyRegionsExample.Program.Main()
        End Sub

        <TestMethod(), Test(), Owner("Console")> _
	        Public Sub XMLMailMerge()
	            TestHelper.SetDataDir("MailMergeAndReporting/XMLMailMerge")
	
	            XMLMailMergeExample.Program.Main()
        End Sub

        <TestMethod(), Test(), Owner("Console")> _
	        Public Sub RemoveEmptyRegions()
	            TestHelper.SetDataDir("MailMergeAndReporting/RemoveEmptyRegions")
	
	            RemoveEmptyRegionsExample.Program.Main()
        End Sub

        <TestMethod(), Test(), Owner("Console")> _
	        Public Sub NestedMailMerge()
	            TestHelper.SetDataDir("MailMergeAndReporting/NestedMailMerge")
	
	            NestedMailMergeExample.Program.Main()
        End Sub

        <TestMethod(), Test(), Owner("Console")> _
	        Public Sub MultipleDocsInMailMerge()
	            TestHelper.SetDataDir("MailMergeAndReporting/MultipleDocsInMailMerge")
	
	            MultipleDocsInMailMergeExample.Program.Main()
        End Sub

        <TestMethod(), Test(), Owner("Console")> _
	        Public Sub MailMergeFormFields()
	            TestHelper.SetDataDir("MailMergeAndReporting/MailMergeFormFields")
	
	            MailMergeFormFieldsExample.Program.Main()
        End Sub

        <TestMethod(), Test(), Owner("Console")> _
	        Public Sub LINQtoXMLMailMerge()
	            TestHelper.SetDataDir("MailMergeAndReporting/LINQtoXMLMailMerge")
	
	            LINQtoXMLMailMergeExample.Program.Main()
        End Sub

	End Class
End Namespace